src/EPR.PRN.ObligationCalculation.Function/StoreApprovedSubmissionsFunction.cs

```diff
@@ -1,33 +1,46 @@
+using EPR.PRN.ObligationCalculation.Application.Configs;
+using EPR.PRN.ObligationCalculation.Application.Services;
 using Microsoft.Azure.Functions.Worker;
+using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.Logging;
 
 namespace EPR.PRN.ObligationCalculation.Function
 {
     public class StoreApprovedSubmissionsFunction
     {
-        private readonly ILogger _logger;
-        private static string LogPrefix = "[StoreApprovedSubmissionsFunction]:";
-        private DateTime? CreatedDate = null;
+        private readonly ISubmissionsDataService _submissionsService;
+        private readonly IAppInsightsProvider _appInsightsProvider;
+        private readonly IServiceBusProvider _serviceBusProvider;
+        private readonly ILogger<StoreApprovedSubmissionsFunction> _logger;
 
-        public StoreApprovedSubmissionsFunction(ILoggerFactory loggerFactory)
+        private readonly string LogPrefix = ApplicationConstants.StoreApprovedSubmissionsFunctionLogPrefix;
+
+        public StoreApprovedSubmissionsFunction(ILogger<StoreApprovedSubmissionsFunction> logger, IConfiguration configuration, ISubmissionsDataService submissionsService, IAppInsightsProvider appInsightsProvider, IServiceBusProvider serviceBusProvider)
         {
-            _logger = loggerFactory.CreateLogger<StoreApprovedSubmissionsFunction>();
+            _logger = logger;
+            _submissionsService = submissionsService;
+            _appInsightsProvider = appInsightsProvider;
+            _serviceBusProvider = serviceBusProvider;
         }
 
-        [Function("Function1")]
-        public void Run([TimerTrigger("0 0 12 * * 1-5")] TimerInfo myTimer)
+        [Function("StoreApprovedSubmissionsFunction")]
+        public async Task RunAsync([TimerTrigger("%StoreApprovedSubmissions:Schedule%")] TimerInfo myTimer)
         {
-            _logger.LogInformation($"{LogPrefix} New session started");
+            _logger.LogInformation("{LogPrefix} >>>>> New session started <<<<< ", LogPrefix);
+
+            var createdDate = _appInsightsProvider.GetParameterForApprovedSubmissionsApiCall().Result;
+            var approvedSubmissionEntities = await _submissionsService.GetApprovedSubmissionsData(createdDate.ToString("yyyy-MM-dd"));
 
-            if (myTimer.ScheduleStatus is not null)
+            if (approvedSubmissionEntities.Count > 0)
```

Comment:
Can this inverted to something like

if (approved... .Count == 0 )
{
_logger.LogInformation("{LogPrefix} COMPLETED >>>>> No new submissions received and send to queue <<<<< ", LogPrefix);
return;
}